### PR TITLE
fix(addon-doc): support coercing value from query params

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes https://github.com/Tinkoff/taiga-ui/issues/1011

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/150551892-22f32dd4-b8ef-46d1-b6f8-f0d487379599.mov

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/addon-doc/src/components/documentation/documentation-property-connector.directive.ts [hunk 1]
```diff
@@ -81,11 +83,11 @@ export class TuiDocDocumentationPropertyConnectorDirective<T>
         return this.documentationPropertyMode !== 'output';
     }
 
-    ngOnChanges() {
+    ngOnChanges(): void {
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-doc/src/components/documentation/documentation-property-connector.directive.ts]: fixed

File: projects/addon-doc/src/components/documentation/documentation-property-connector.directive.ts [hunk 2]
```diff
@@ -85,55 +87,45 @@ export class TuiDocDocumentationPropertyConnectorDirective<T>
...
-        const isValueAvailableByKey =
-            !!params[this.documentationPropertyName + SERIALIZED_SUFFIX];
-
-        if (isValueAvailableByKey && this.documentationPropertyValues) {
-            this.onValueChange(
-                this.documentationPropertyValues[
-                    params[this.documentationPropertyName + SERIALIZED_SUFFIX]
-                ],
-            );
+    private parseParams(params: Params): void {
```
Comment (vladimirpotekhin) [inline projects/addon-doc/src/components/documentation/documentation-property-connector.directive.ts:103]: nit: redundant "void"

File: projects/addon-doc/src/utils/coerce-value.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    return decodeURIComponent(prepared);
+}
+
+function isEmptyParamValue(value: string): boolean {
+    return ['undefined', 'null', 'NaN', 'Infinity'].includes(value);
+}
+
+function isBooleanParamValue(value: string): boolean {
+    return String(value) === 'true' || String(value) === 'false';
```
Comment (vladimirpotekhin) [inline projects/addon-doc/src/utils/coerce-value.ts:20]: should't we remove String() here? value is already string

**Review by waterplea — APPROVED** [review]

File: projects/addon-doc/src/utils/coerce-value.ts (hunk 3 above)
Comment (splincode) [reply projects/addon-doc/src/utils/coerce-value.ts]: fixed

File: projects/addon-doc/src/components/documentation/documentation-property-connector.directive.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-doc/src/components/documentation/documentation-property-connector.directive.ts]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**